src/dali_executor/dali_executor.cc

```diff
@@ -26,35 +26,45 @@
 
 namespace triton { namespace backend { namespace dali {
 
-std::vector<int>
-distribute_batch_size(int batch_size)
+template <bool owns>
+void
+SetupInputs(DaliPipeline &pipeline, const std::vector<IODescr<owns>>& inputs)
 {
-  std::vector<int> result;
-  for (int part = 1; batch_size > 0; part *= 2, batch_size /= 2) {
-    if (batch_size % 2)
-      result.push_back(part);
+  assert(!inputs.empty());
+  int batch_size = inputs[0].shape.num_samples();
+  for (size_t i = 1; i < inputs.size(); ++i) {
+    assert(
+        inputs[i].shape.num_samples() == batch_size &&
+        "All inputs should have equal batch size.");
+  }
+  for (auto& inp : inputs) {
+    assert(
+        inp.shape.num_elements() * dali_type_size(inp.type) <=
+        inp.buffer.size());
```

Comment:
Is this line break really required? This looks weird.

Reply:
We use clang-format used across all of the triton-server code.